### DISCUSSION TITLE
Import 3D mesh

### DISCUSSION THREAD

**Discussion by ghost** [AMReX-Combustion/PeleLMeX#418] [Q&A]
Hello, 

I have a 3D mesh of a combustion chamber. I want to import it in PeleLMex. Please help me.  Is there any entry I can make in the input file? 

Thank you

**Comment by drummerdoc** [comment]
You can look into AMReX's capabilities with STL files.  You will likely run into issues being able to coarsen the geometry to produce a sufficient number of levels for the multigrid solver to perform well, depending on the complexity of your geometry.  However, we are in the process of adding an AMG-based solver capability to PeleLMeX which may help with that aspect.  Perhaps you can create a fork of the repo and include the geometry information and forward a link?  If so, we can try to include your case in our testing.

**Comment by baperry2** [comment]
A few notes beyond what @drummerdoc said:

- Right now you'll probably run into issues with the addFineLevels function not being supported for STL geometries. See the linked AMReX pull request for details on how to deal with that. 
- You can see the example STL case that was added in PeleC, the setup in LMeX should be similar: https://github.com/AMReX-Combustion/PeleC/pull/771
- Complex geometries using STL files are an ongoing area of development and may not be stable yet. Building up from basic shapes using AMReX EB functions may be easier if possible

**Comment by ghost** [comment]
@drummerdoc  Thank you sir,  the case is :- I have a Cylindrical shape with with smaller cylinders attached to it, from where the CH4 and pilot flame enters.  If you are in the process of  adding the geometry  capabilities please look into the .3mf format (https://3mf.io/). I have been trying to add this into OpenFOAM. This format does not support non-orthogonal edges by design.  I will add the geometry and create the pull request.  

@baperry2  Oh! There are two codes, which one is better for diffusion flames and spray combustion?? I will try to perform my case in PeleC too.

**Comment by baperry2** [comment]
If your geometry is just a combination of different cylinders, you should definitely just use the capability to build your geometry up from simple shapes, see documentation: 

- https://amrex-combustion.github.io/PeleLMeX/manual/html/LMeXControls.html#embedded-geometry
- https://amrex-codes.github.io/amrex/docs_html/EB.html

Both codes can support that use case. As the names imply, LMeX if preferred for low Mach number cases and C is preferred for fully compressible cases, although there are some other subtleties in specific capability differences between the codes.
